src/Microsoft.DotNet.Darc/DarcLib/Helpers/NullSafeUnixPathDictionary.cs

```diff
@@ -0,0 +1,51 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+
+using System.Collections.Generic;
+
+#nullable enable
+namespace Microsoft.DotNet.DarcLib.Helpers;
+
+/// <summary>
+/// A dictionary that treats null UnixPath keys as equivalent to UnixPath(".") for root directory operations.
+/// This allows using null to represent the root directory while maintaining dictionary functionality.
+/// </summary>
+/// <typeparam name="TValue">The type of values in the dictionary</typeparam>
+public class NullSafeUnixPathDictionary<TValue> : IEnumerable<KeyValuePair<UnixPath?, TValue>>
+{
+    private readonly Dictionary<UnixPath, TValue> _inner = new();
+    private static readonly UnixPath RootPath = new(".");
+
+    private static UnixPath NormalizeKey(UnixPath? key) 
+    {
+        if (key == null) return RootPath;
+        
+        var path = key.ToString();
+        return path == "/" || path == "" || path == "." ? RootPath : key;
```

Comment:
Btw this would be allocation-free:
```suggestion
        return (path.Length == 0 || (path.Length == 1 && path[0] == '.')) ? RootPath : key;
```